### ISSUE TITLE
[PPC0022] - Initial entry point as functions vs. constructor-methods

### ISSUE THREAD

**Issue by leonerd** [Perl/PPCs#44]
The original design had functions such as

```perl
my $metapkg = meta::get_package( "some::name::here" );
```

A suggestion from the mailing list was that these could instead be written as constructor-style methods on named packages within `meta`:

```perl
my $metapkg = meta::package->get("some::name::here");
```

Having implemented both now it feels actually a little nicer to use the latter style. Plus it conveniently gives a place to explain and document what actual class these meta-objects appear to be in.

If we think this style is better we should update the doc to match.

**Comment by rwp0** [comment]
Yeah, +1, I always disliked `get_OBJECT`, `set_OBJECT` language, the shorter the better.
